{"cells":[{"cell_type":"code","execution_count":null,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":4,"status":"ok","timestamp":1711540108846,"user":{"displayName":"Sicheng Liu","userId":"13381977535991759905"},"user_tz":-480},"id":"CfarDRogRPsn","outputId":"fd421507-2514-4160-9c04-6b88bdd244e4"},"outputs":[{"data":{"text/plain":["['kmeans_model.pkl',\n"," 'kmeans_model_100.pkl',\n"," 'haarcascade_frontalface_default.xml',\n"," 'decode.py',\n"," 'utils_tokenizer.py',\n"," 'utils.py',\n"," 'model_generator.py',\n"," 'data',\n"," 'Untitled0.ipynb',\n"," '__pycache__',\n"," 'multimodal_model',\n"," 'models',\n"," 'runs',\n"," 'output3.15',\n"," 'output3.17',\n"," 'util_dataset.py',\n"," 'output3.18',\n"," 'output3.19',\n"," 'output3.21',\n"," 'output3.22',\n"," 'clipsum.py',\n"," 'train.py',\n"," 'Untitled1.ipynb',\n"," 'output3.26',\n"," 'output']"]},"execution_count":1,"metadata":{},"output_type":"execute_result"}],"source":["import os\n","path=\"/content/drive/MyDrive/Research\"\n","os.chdir(path)\n","os.listdir(path)"]},{"cell_type":"code","execution_count":null,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":13797,"status":"ok","timestamp":1711540122641,"user":{"displayName":"Sicheng Liu","userId":"13381977535991759905"},"user_tz":-480},"id":"fK6LvP6yaXig","outputId":"9c5f19f5-98bb-4927-c0f4-0adda48421b3"},"outputs":[{"name":"stdout","output_type":"stream","text":["Collecting transformers==4.27.4\n","  Downloading transformers-4.27.4-py3-none-any.whl (6.8 MB)\n","\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m6.8/6.8 MB\u001b[0m \u001b[31m31.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hRequirement already satisfied: filelock in /usr/local/lib/python3.10/dist-packages (from transformers==4.27.4) (3.13.1)\n","Requirement already satisfied: huggingface-hub<1.0,>=0.11.0 in /usr/local/lib/python3.10/dist-packages (from transformers==4.27.4) (0.20.3)\n","Requirement already satisfied: numpy>=1.17 in /usr/local/lib/python3.10/dist-packages (from transformers==4.27.4) (1.25.2)\n","Requirement already satisfied: packaging>=20.0 in /usr/local/lib/python3.10/dist-packages (from transformers==4.27.4) (24.0)\n","Requirement already satisfied: pyyaml>=5.1 in /usr/local/lib/python3.10/dist-packages (from transformers==4.27.4) (6.0.1)\n","Requirement already satisfied: regex!=2019.12.17 in /usr/local/lib/python3.10/dist-packages (from transformers==4.27.4) (2023.12.25)\n","Requirement already satisfied: requests in /usr/local/lib/python3.10/dist-packages (from transformers==4.27.4) (2.31.0)\n","Collecting tokenizers!=0.11.3,<0.14,>=0.11.1 (from transformers==4.27.4)\n","  Downloading tokenizers-0.13.3-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (7.8 MB)\n","\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m7.8/7.8 MB\u001b[0m \u001b[31m85.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hRequirement already satisfied: tqdm>=4.27 in /usr/local/lib/python3.10/dist-packages (from transformers==4.27.4) (4.66.2)\n","Requirement already satisfied: fsspec>=2023.5.0 in /usr/local/lib/python3.10/dist-packages (from huggingface-hub<1.0,>=0.11.0->transformers==4.27.4) (2023.6.0)\n","Requirement already satisfied: typing-extensions>=3.7.4.3 in /usr/local/lib/python3.10/dist-packages (from huggingface-hub<1.0,>=0.11.0->transformers==4.27.4) (4.10.0)\n","Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.10/dist-packages (from requests->transformers==4.27.4) (3.3.2)\n","Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.10/dist-packages (from requests->transformers==4.27.4) (3.6)\n","Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.10/dist-packages (from requests->transformers==4.27.4) (2.0.7)\n","Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.10/dist-packages (from requests->transformers==4.27.4) (2024.2.2)\n","Installing collected packages: tokenizers, transformers\n","  Attempting uninstall: tokenizers\n","    Found existing installation: tokenizers 0.15.2\n","    Uninstalling tokenizers-0.15.2:\n","      Successfully uninstalled tokenizers-0.15.2\n","  Attempting uninstall: transformers\n","    Found existing installation: transformers 4.38.2\n","    Uninstalling transformers-4.38.2:\n","      Successfully uninstalled transformers-4.38.2\n","Successfully installed tokenizers-0.13.3 transformers-4.27.4\n"]}],"source":["!pip install transformers==4.27.4"]},{"cell_type":"code","execution_count":null,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":16405,"status":"ok","timestamp":1711540139038,"user":{"displayName":"Sicheng Liu","userId":"13381977535991759905"},"user_tz":-480},"id":"GAlAE3u7RzI6","outputId":"aef4adc8-7968-4012-8721-bcc496afe388"},"outputs":[{"name":"stdout","output_type":"stream","text":["Collecting scenedetect\n","  Downloading scenedetect-0.6.3-py3-none-any.whl (114 kB)\n","\u001b[?25l     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m0.0/114.7 kB\u001b[0m \u001b[31m?\u001b[0m eta \u001b[36m-:--:--\u001b[0m\r\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m114.7/114.7 kB\u001b[0m \u001b[31m3.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hRequirement already satisfied: Click in /usr/local/lib/python3.10/dist-packages (from scenedetect) (8.1.7)\n","Requirement already satisfied: numpy in /usr/local/lib/python3.10/dist-packages (from scenedetect) (1.25.2)\n","Requirement already satisfied: platformdirs in /usr/local/lib/python3.10/dist-packages (from scenedetect) (4.2.0)\n","Requirement already satisfied: tqdm in /usr/local/lib/python3.10/dist-packages (from scenedetect) (4.66.2)\n","Installing collected packages: scenedetect\n","Successfully installed scenedetect-0.6.3\n","Collecting POT\n","  Downloading POT-0.9.3-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (823 kB)\n","\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m823.0/823.0 kB\u001b[0m \u001b[31m13.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hRequirement already satisfied: numpy>=1.16 in /usr/local/lib/python3.10/dist-packages (from POT) (1.25.2)\n","Requirement already satisfied: scipy>=1.6 in /usr/local/lib/python3.10/dist-packages (from POT) (1.11.4)\n","Installing collected packages: POT\n","Successfully installed POT-0.9.3\n","Collecting datasets\n","  Downloading datasets-2.18.0-py3-none-any.whl (510 kB)\n","\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m510.5/510.5 kB\u001b[0m \u001b[31m11.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hRequirement already satisfied: filelock in /usr/local/lib/python3.10/dist-packages (from datasets) (3.13.1)\n","Requirement already satisfied: numpy>=1.17 in /usr/local/lib/python3.10/dist-packages (from datasets) (1.25.2)\n","Requirement already satisfied: pyarrow>=12.0.0 in /usr/local/lib/python3.10/dist-packages (from datasets) (14.0.2)\n","Requirement already satisfied: pyarrow-hotfix in /usr/local/lib/python3.10/dist-packages (from datasets) (0.6)\n","Collecting dill<0.3.9,>=0.3.0 (from datasets)\n","  Downloading dill-0.3.8-py3-none-any.whl (116 kB)\n","\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m116.3/116.3 kB\u001b[0m \u001b[31m11.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hRequirement already satisfied: pandas in /usr/local/lib/python3.10/dist-packages (from datasets) (1.5.3)\n","Requirement already satisfied: requests>=2.19.0 in /usr/local/lib/python3.10/dist-packages (from datasets) (2.31.0)\n","Requirement already satisfied: tqdm>=4.62.1 in /usr/local/lib/python3.10/dist-packages (from datasets) (4.66.2)\n","Collecting xxhash (from datasets)\n","  Downloading xxhash-3.4.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (194 kB)\n","\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m194.1/194.1 kB\u001b[0m \u001b[31m16.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hCollecting multiprocess (from datasets)\n","  Downloading multiprocess-0.70.16-py310-none-any.whl (134 kB)\n","\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m134.8/134.8 kB\u001b[0m \u001b[31m13.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hRequirement already satisfied: fsspec[http]<=2024.2.0,>=2023.1.0 in /usr/local/lib/python3.10/dist-packages (from datasets) (2023.6.0)\n","Requirement already satisfied: aiohttp in /usr/local/lib/python3.10/dist-packages (from datasets) (3.9.3)\n","Requirement already satisfied: huggingface-hub>=0.19.4 in /usr/local/lib/python3.10/dist-packages (from datasets) (0.20.3)\n","Requirement already satisfied: packaging in /usr/local/lib/python3.10/dist-packages (from datasets) (24.0)\n","Requirement already satisfied: pyyaml>=5.1 in /usr/local/lib/python3.10/dist-packages (from datasets) (6.0.1)\n","Requirement already satisfied: aiosignal>=1.1.2 in /usr/local/lib/python3.10/dist-packages (from aiohttp->datasets) (1.3.1)\n","Requirement already satisfied: attrs>=17.3.0 in /usr/local/lib/python3.10/dist-packages (from aiohttp->datasets) (23.2.0)\n","Requirement already satisfied: frozenlist>=1.1.1 in /usr/local/lib/python3.10/dist-packages (from aiohttp->datasets) (1.4.1)\n","Requirement already satisfied: multidict<7.0,>=4.5 in /usr/local/lib/python3.10/dist-packages (from aiohttp->datasets) (6.0.5)\n","Requirement already satisfied: yarl<2.0,>=1.0 in /usr/local/lib/python3.10/dist-packages (from aiohttp->datasets) (1.9.4)\n","Requirement already satisfied: async-timeout<5.0,>=4.0 in /usr/local/lib/python3.10/dist-packages (from aiohttp->datasets) (4.0.3)\n","Requirement already satisfied: typing-extensions>=3.7.4.3 in /usr/local/lib/python3.10/dist-packages (from huggingface-hub>=0.19.4->datasets) (4.10.0)\n","Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.10/dist-packages (from requests>=2.19.0->datasets) (3.3.2)\n","Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.10/dist-packages (from requests>=2.19.0->datasets) (3.6)\n","Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.10/dist-packages (from requests>=2.19.0->datasets) (2.0.7)\n","Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.10/dist-packages (from requests>=2.19.0->datasets) (2024.2.2)\n","Requirement already satisfied: python-dateutil>=2.8.1 in /usr/local/lib/python3.10/dist-packages (from pandas->datasets) (2.8.2)\n","Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.10/dist-packages (from pandas->datasets) (2023.4)\n","Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.10/dist-packages (from python-dateutil>=2.8.1->pandas->datasets) (1.16.0)\n","Installing collected packages: xxhash, dill, multiprocess, datasets\n","Successfully installed datasets-2.18.0 dill-0.3.8 multiprocess-0.70.16 xxhash-3.4.1\n"]}],"source":["!pip install scenedetect\n","!pip install POT\n","!pip install datasets"]},{"cell_type":"code","execution_count":1,"metadata":{"colab":{"background_save":true,"base_uri":"https://localhost:8080/"},"id":"ztt0ShRmRmZX"},"outputs":[{"name":"stdout","output_type":"stream","text":["^C\n"]},{"name":"stderr","output_type":"stream","text":["Traceback (most recent call last):\n","  File \"d:\\Anaconda\\envs\\assignment2\\lib\\site-packages\\transformers\\utils\\import_utils.py\", line 1126, in _get_module\n","    return importlib.import_module(\".\" + module_name, self.__name__)\n","  File \"d:\\Anaconda\\envs\\assignment2\\lib\\importlib\\__init__.py\", line 127, in import_module\n","    return _bootstrap._gcd_import(name[level:], package, level)\n","  File \"<frozen importlib._bootstrap>\", line 1014, in _gcd_import\n","  File \"<frozen importlib._bootstrap>\", line 991, in _find_and_load\n","  File \"<frozen importlib._bootstrap>\", line 975, in _find_and_load_unlocked\n","  File \"<frozen importlib._bootstrap>\", line 671, in _load_unlocked\n","  File \"<frozen importlib._bootstrap_external>\", line 843, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n","  File \"d:\\Anaconda\\envs\\assignment2\\lib\\site-packages\\transformers\\models\\clip\\feature_extraction_clip.py\", line 20, in <module>\n","    from .image_processing_clip import CLIPImageProcessor\n","  File \"d:\\Anaconda\\envs\\assignment2\\lib\\site-packages\\transformers\\models\\clip\\image_processing_clip.py\", line 22, in <module>\n","    from ...image_transforms import (\n","  File \"d:\\Anaconda\\envs\\assignment2\\lib\\site-packages\\transformers\\image_transforms.py\", line 48, in <module>\n","    import tensorflow as tf\n","  File \"d:\\Anaconda\\envs\\assignment2\\lib\\site-packages\\tensorflow\\__init__.py\", line 37, in <module>\n","    from tensorflow.python.tools import module_util as _module_util\n","  File \"d:\\Anaconda\\envs\\assignment2\\lib\\site-packages\\tensorflow\\python\\__init__.py\", line 37, in <module>\n","    from tensorflow.python.eager import context\n","  File \"d:\\Anaconda\\envs\\assignment2\\lib\\site-packages\\tensorflow\\python\\eager\\context.py\", line 29, in <module>\n","    from tensorflow.core.framework import function_pb2\n","  File \"d:\\Anaconda\\envs\\assignment2\\lib\\site-packages\\tensorflow\\core\\framework\\function_pb2.py\", line 5, in <module>\n","    from google.protobuf.internal import builder as _builder\n","ImportError: cannot import name 'builder' from 'google.protobuf.internal' (d:\\Anaconda\\envs\\assignment2\\lib\\site-packages\\google\\protobuf\\internal\\__init__.py)\n","\n","The above exception was the direct cause of the following exception:\n","\n","Traceback (most recent call last):\n","  File \"train.py\", line 14, in <module>\n","    from clipsum import CLIPSum\n","  File \"d:\\Study\\Master Document\\Research\\Research Pathway\\code\\clipsum.py\", line 447, in <module>\n","    from transformers import CLIPFeatureExtractor, CLIPModel, CLIPTokenizer, CLIPTextModel\n","  File \"<frozen importlib._bootstrap>\", line 1039, in _handle_fromlist\n","  File \"d:\\Anaconda\\envs\\assignment2\\lib\\site-packages\\transformers\\utils\\import_utils.py\", line 1117, in __getattr__\n","    value = getattr(module, name)\n","  File \"d:\\Anaconda\\envs\\assignment2\\lib\\site-packages\\transformers\\utils\\import_utils.py\", line 1116, in __getattr__\n","    module = self._get_module(self._class_to_module[name])\n","  File \"d:\\Anaconda\\envs\\assignment2\\lib\\site-packages\\transformers\\utils\\import_utils.py\", line 1128, in _get_module\n","    raise RuntimeError(\n","RuntimeError: Failed to import transformers.models.clip.feature_extraction_clip because of the following error (look up to see its traceback):\n","cannot import name 'builder' from 'google.protobuf.internal' (d:\\Anaconda\\envs\\assignment2\\lib\\site-packages\\google\\protobuf\\internal\\__init__.py)\n"]}],"source":["# !python train.py --epochs 1 --batch_size 2 --save_model_path 'multimodal_model/2024-Mar-21-08-28-10/2024-Mar-22-14-46-14/' --model_name 'E0- 0.556437.ckpt' --resume_training Ture\n","!python train.py --epochs 1 --batch_size 2\n"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"H3IGmln00i6m"},"outputs":[],"source":[]},{"cell_type":"code","execution_count":null,"metadata":{"id":"S7ooWhX7aoGr"},"outputs":[],"source":["from google.colab import runtime\n","runtime.unassign()"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"1ea3eNyrcTNF"},"outputs":[],"source":[]}],"metadata":{"accelerator":"GPU","colab":{"authorship_tag":"ABX9TyN1p1gjji89N7JRxANehR5U","gpuType":"V100","machine_shape":"hm","mount_file_id":"1oMXo26krS520rIIH9N290TnIN3ZZeuRg","name":"","version":""},"kernelspec":{"display_name":"Python 3","name":"python3"},"language_info":{"codemirror_mode":{"name":"ipython","version":3},"file_extension":".py","mimetype":"text/x-python","name":"python","nbconvert_exporter":"python","pygments_lexer":"ipython3","version":"3.8.18"}},"nbformat":4,"nbformat_minor":0}
